test/entropic.jl

```diff
@@ -80,52 +103,51 @@ Random.seed!(100)
-            @test all([
-                isapprox(Float32.(γ_pot[i]), γ_all[:, :, i]; rtol=1e-3) for i in 1:d
-            ])
-            @test eltype(γ_all) == Float32
+            for (size2_μ, size2_ν) in
+                (((), (d,)), ((1,), (d,)), ((d,), ()), ((d,), (1,)), ((d,), (d,)))
+                # generate uniform histograms
+                μ = fill(Float32(1 / M), (M, size2_μ...))
+                ν = fill(Float32(1 / N), (N, size2_ν...))
+
+                # compute optimal transport plan and check that it is consistent with the
+                # plan for individual histograms
+                γ_all = sinkhorn(μ, ν, C, eps; maxiter=5_000, rtol=1e-6)
+                @test eltype(γ_all) === Float32
+                @test size(γ_all) == (M, N, d)
+                @test all(view(γ_all, :, :, i) ≈ γ for i in axes(γ_all, 3))
+
+                # compute optimal transport cost and check that it is consistent with the
+                # cost for individual histograms
+                c_all = sinkhorn2(μ, ν, C, eps; maxiter=5_000, rtol=1e-6)
+                @test eltype(c_all) === Float32
+                @test size(c_all) == (d,)
+                @test all(x ≈ c for x in c_all)
+            end
         end
 
-        @testset "batch" begin
-            # create two sets of batch histograms 
-            d = 10
-            μ = rand(Float64, (M, d))
-            μ = μ ./ sum(μ; dims=1)
-            ν = rand(Float64, (N, d))
-            ν = ν ./ sum(ν; dims=1)
-
-            # create random cost matrix
+        # https://github.com/JuliaOptimalTransport/OptimalTransport.jl/issues/86
+        @testset "AD" begin
+            # uniform histograms with random cost matrix
+            μ = fill(1 / M, M)
+            ν = fill(1 / N, N)
             C = pairwise(SqEuclidean(), rand(1, M), rand(1, N); dims=2)
 
-            # compute optimal transport map (Julia implementation + POT)
-            eps = 0.01
-            γ_all = sinkhorn(μ, ν, C, eps; maxiter=5_000)
-            γ_pot = [POT.sinkhorn(μ[:, i], ν[:, i], C, eps; numItermax=5_000) for i in 1:d]
-            @test all([isapprox(γ_all[:, :, i], γ_pot[i]; rtol=1e-6) for i in 1:d])
-
-            c_all = sinkhorn2(μ, ν, C, eps; maxiter=5_000)
-            c_pot = [
-                POT.sinkhorn2(μ[:, i], ν[:, i], C, eps; numItermax=5_000)[1] for i in 1:d
-            ]
-            @test c_all ≈ c_pot rtol = 1e-6
-
-            γ_all = sinkhorn(μ[:, 1], ν, C, eps; maxiter=5_000)
-            γ_pot = [POT.sinkhorn(μ[:, 1], ν[:, i], C, eps; numItermax=5_000) for i in 1:d]
-            @test all([isapprox(γ_all[:, :, i], γ_pot[i]; rtol=1e-6) for i in 1:d])
-
-            γ_all = sinkhorn(μ, ν[:, 1], C, eps; maxiter=5_000)
-            γ_pot = [POT.sinkhorn(μ[:, i], ν[:, 1], C, eps; numItermax=5_000) for i in 1:d]
-            @test all([isapprox(γ_all[:, :, i], γ_pot[i]; rtol=1e-6) for i in 1:d])
+            # compute gradients with respect to source and target marginals separately and
```

Comment:
I'll see if I can make a separate PR to check the correctness of these gradients.